### PR TITLE
CDS-57454 AWS Backoff Strategy Settings

### PR DESCRIPTION
**Pull request by michaelcretzman**
https://harness.atlassian.net/browse/CDS-57454

# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ x] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* https://harness.atlassian.net/browse/CDS-57454

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: docs/platform/7_Connectors/Cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md [hunk 1]
```diff
@@ -643,14 +643,15 @@ For more information about installing software with the delegate, go to [Build c
...
   * **AWS Access Key:** The [Access Key and Secret Access Key](https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html#access-keys-and-secret-access-keys) of the IAM Role to use for the AWS account. You can use [Harness Text Secrets](../../../Secrets/2-add-use-text-secrets.md) for both.
   * **Use IRSA:** Allows the Harness Kubernetes delegate in AWS EKS to use a specific IAM role when making authenticated requests to resources. By default, the Harness Kubernetes delegate uses a ClusterRoleBinding to the **default** service account; whereas, with this option, you can use AWS [IAM roles for service accounts (IRSA)](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) to associate a specific IAM role with the service account used by the Harness Kubernetes delegate.
+* **AWS Backoff Strategy:** see [AWS Backoff Strategy](#aws-backoff-strategy) below.
```
Comment (brian-f-harness) [inline docs/platform/7_Connectors/Cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md:654]:
```suggestion
* **AWS Backoff Strategy:** Go to [AWS Backoff Strategy](#aws-backoff-strategy) below.
```

File: docs/platform/7_Connectors/Cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md [hunk 2]
```diff
@@ -757,6 +758,85 @@ If you want to use an AWS GovCloud account for this connector, select it in the
...
+"RequestLimitExceeded",  
+"BandwidthLimitExceeded",  
+"RequestThrottled",  
+"RequestThrottledException",  
+"EC2ThrottledException",  
+"PriorRequestNotComplete",  
+"429 Too Many Requests"
+```
+
+For more strategies, see [Exponential Backoff And Jitter](https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/) from AWS.
```
Comment (brian-f-harness) [inline docs/platform/7_Connectors/Cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md:824]:
```suggestion
For more strategies, go to [Exponential Backoff And Jitter](https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/) from AWS.
```

File: docs/platform/7_Connectors/Cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md [hunk 3]
```diff
@@ -757,6 +758,85 @@ If you want to use an AWS GovCloud account for this connector, select it in the
...
+
+#### Setting the AWS backoff strategy
+
+When you create a Harness AWS connector, you can use the backoff strategy settings to configure the AWS backoff strategy.
+
+These options are part of the AWS [software.amazon.awssdk.core.retry.backoff](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/core/retry/backoff/package-summary.html) package.
+
+The settings are:
+
+- **Fixed Delay:** this is a simple backoff strategy that always uses a [fixed delay](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/core/retry/backoff/FixedDelayBackoffStrategy.html) before the next retry attempt.
```
Comment (brian-f-harness) [inline docs/platform/7_Connectors/Cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md:834]:
```suggestion
- **Fixed Delay:** This is a simple backoff strategy that always uses a [fixed delay](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/core/retry/backoff/FixedDelayBackoffStrategy.html) before the next retry attempt.
```

File: docs/platform/7_Connectors/Cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md [hunk 4]
```diff
@@ -757,6 +758,85 @@ If you want to use an AWS GovCloud account for this connector, select it in the
...
+- **Equal Jitter:** this strategy uses [equal jitter](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/core/retry/backoff/EqualJitterBackoffStrategy.html) for computing the delay before the next retry.
```
Comment (brian-f-harness) [inline docs/platform/7_Connectors/Cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md:835]:
```suggestion
- **Equal Jitter:** This strategy uses [equal jitter](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/core/retry/backoff/EqualJitterBackoffStrategy.html) for computing the delay before the next retry.
```

File: docs/platform/7_Connectors/Cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md [hunk 5]
```diff
@@ -757,6 +758,85 @@ If you want to use an AWS GovCloud account for this connector, select it in the
...
+- **Full Jitter:** this strategy uses a [full jitter](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/core/retry/backoff/FullJitterBackoffStrategy.html) strategy for computing the next backoff delay.
```
Comment (brian-f-harness) [inline docs/platform/7_Connectors/Cloud-providers/ref-cloud-providers/aws-connector-settings-reference.md:836]:
```suggestion
- **Full Jitter:** This strategy uses a [full jitter](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/core/retry/backoff/FullJitterBackoffStrategy.html) strategy for computing the next backoff delay.  
```

**Review by brian-f-harness — APPROVED** [review]
lgtm. Added a couple minor suggestions.

**Review by shylajasun — APPROVED** [review]

**Pull request merged**